src/_pytest/compat.py

```diff
@@ -162,7 +162,7 @@ def getfuncargnames(
     # it's passed as an unbound method or function, remove the first
     # parameter name.
     if is_method or (
-        cls and not isinstance(cls.__dict__.get(name, None), staticmethod)
+        cls and not isinstance(inspect.getattr_static(cls, name), staticmethod)
```

Comment:
Since the `getattr_static` is a bit subtle, I suggest adding a short comment about it, something like:

```py
# Not using getattr because we don't want to resolve the staticmethod.
# Not using cls.__dict__ because we want to check the entire MRO.
```

Reply:
added in 0bf451b - thanks @bluetech